### ISSUE TITLE
[BUG]: Error “Instellen mislukt, wordt opnieuw geprobeerd: Data incomplete or missing” in home assistant

### ISSUE THREAD

**Issue by Torch1969** [plugwise/plugwise-beta#739]
### Describe the bug.

Reference discussion with @bouwew at tweakers: https://gathering.tweakers.net/forum/list_message/80294530#80294530

I’m using the Plugwise core integration
History:
In Plugwise app: Moved a plug to another room and renamed it
In home assistant: the renamed plug isn’t updated, and entities are not available anymore
In Plugwise app: restarted the Adam
In home asssitant: I removed the Plugwise integration an added it again (suggestion from @bouwew)
When adding my Adam hub again I get the error message as stated in the title. It is repeating with “initialiseren” and then the error message over and over again.

Debug log from HA:
2024-10-03 12:21:29.091 DEBUG (MainThread) [homeassistant.components.plugwise] Finished fetching plugwise data in 1.127 seconds (success: False)
2024-10-03 12:21:28.354 DEBUG (MainThread) [plugwise.constants] Plugwise identified as smile_open_therm_v3

I cannot download diagnostic info, because Adam device is not added.

### What version of Plugwise Beta are you using?

Core version 2024.9

### What was the last working version of Plugwise Beta (or Core if not using beta before)?

Core version 2024.9

### What version of Home Assistant Core are you using?

Core 2024.9

### What type of installation are you running?

Home Assistant Container

### How did you install plugwise-beta?

Home Assistant Core

### What kind of Plugwise device is having issues?

Smile: Adam (including Lisa, Tom, Floor)

### What firmware version is your Plugwise product at?

3.7.8

### Logging

No log lines.

### Diagnostics information

The device isn’t added, so no diagnostic information

**Comment by bouwew** [comment]
Hi @Torch1969 I did some more code analysis, this error is a bit strange.
This error would only show up when the initial connection, made during the adding-process of the Integration, was successful. And then the next attempts to request more data from the Adam, all fail.

Are there any other errors or warnings in the HA LOGBOOK that not directly seem related to Plugwise? Nowadays HA also shows errors/warnings that are not related to any Integration directly but when looked at, show a list of potential issues with various integrations.

Also, could it be that the WiFi-connection to the Adam is not so good? That might explain why the initial connection succeeds but the regular requests for updated data fails. Although I then would expect that sometimes the connection is there.

Could it be that the IP-address of the Adam has changed? If possible, assign a fixed address to the Adam in your router.

**Comment by bouwew** [comment]
Another question: how did you add the Plugwise integration? Via the discovered `Plugwise Adam v3.7.8` button shown at the Setting --> Devices & Services menu? Or via the manual method, pressing the `+Add Integration` button?

**Comment by Torch1969** [comment]
Hi @bouwew thx for looking into this.
I will send you a log from the moment after restart of HA yesterday afternoon and some attempts to add Plugwise integration.
The Adam has a fixed ip address and stable WiFi connection which never gave any problems in the past (2 years and running connection to a homey and a few weeks to home assistant.) this problem started after renaming an moving a plug in Plugwise.
I tried both variants for adding the the integration. After discovery by home assistant and adding the integration myself. The Adam hub is found, added and shown, but it seems that the loading of the configuration fails. The Adam hub isn’t shown as a device in the devices list.
![Schermafbeelding 2024-10-05 om 10 46 21](https://github.com/user-attachments/assets/e40d8322-9812-4fb4-b1d8-beba9286ea5b)

Is there some way to intercept the configuration json from my Adam? I expect there is something strange in there?

**Comment by bouwew** [comment]
I'm already discussing with @CoMPaTech on what could be going on. We're not sure yet what is happening.

Ok, good to know that the Adam is discovered by HA, and that the Adam is also found during the adding-process.
But step 2 of adding the integration, collecting the detailed data from the Adam fails for some reason.

Can you try connecting to this URL: `http://<ip of Adam>/core/domain_objects`, username is `smile`, password is the Smile ID. You should see a long page with xml-data. Google Chrome works best to show this.

On that page there should be a few lines like these:
```
<appliance id="da224107914542988a88561b4452b0f6">
<name>Gateway</name>
```
Do you see these in your browser?

**Comment by Torch1969** [comment]
I can open that page. I don’t use Google Chrome, but in Safari on iPad I can see the text “Gateway Container  for variables …..”
`Gateway Container for variables logged about the Gateway in general. gateway 2022-06-27T10:15:48.537+02:00 2024-10-05T11:34:59.903+02:00 lan_ip_address 2024-10-05T11:34:59+02:00 2024-10-05T11:34:59+02:00 0.0.0.0 regulation_mode 2022-10-21T16:17:50.538+02:00 2022-10-18T20:31:24.937+02:00 heating wlan_ip_address 2024-10-05T11:34:59+02:00 2024-10-05T11:34:59+02:00 192.168.1.104 lan_state 2024-10-05T11:34:59.842+02:00 2024-10-05T11:34:59.842+02:00 down gateway_mode signal_strength dBm 2024-10-05T11:34:59.863+02:00 2024-10-05T11:34:59.863+02:00 -50.00 wlan_state 2024-10-05T11:34:59.864+02:00 2024-10-05T11:34:59.864+02:00 up link_quality 2024-10-05T11:34:59.863+02:00 2024-10-05T11:34:59.863+02:00 60 gateway_mode full full setup light vacation secure away 2022-10-21T16:17:50.538+02:00 regulation_mode heating bleeding_hot bleeding_cold heating off  `

**Comment by bouwew** [comment]
OK that looks good. Then adding the integration to HA should work.
Can you try once more? First, delete the Plugwise Adam integration via the menu "behind" the 3-vertical-points, then restart HA, then re-add the Plugwise integration.

If that still doesn't work, then please install the Plugwise-beta custom_component (for instance via HACS, see here: https://github.com/plugwise/plugwise-beta#how-to-install
This will allow us to troubleshoot in a much easier and quicker way.
Plugwise Beta is basically the same as HA Plugwise. All the changes/improvements we put into Plugwise Beta will end up sometime later in HA Plugwise.

Once installed, please delete the failing Plugwise Adam integration again, restart HA, and then add the Adam manually via selecting the Plugwise Beta integration. And let us know the result here.

**Comment by bouwew** [comment]
Also, for extra debugging please add one extra line into `configuation.yaml`:

```
logger:
  default: warning
  logs:
    ...
    plugwise.constants: debug  <-- add this line
```
This will show the debug and error-messages from the plugwise backend library.

**Comment by bouwew** [comment]
Please add the line `plugwise.constants: debug` into `configuation.yaml` first, before you perform the "try once more" from two messages above.
It that retry fails again, please check the HA LOGBOOK for messages from `plugwise.constants`. If present this/these message(s) might tell us more.

**Comment by Torch1969** [comment]
Ok, both scenarios same error.
Deleting Plugwise integration, restart home asssitant, adding found Adam with Plugwise integration.

after that I removed Plugwise integration and installed Plugwise-beta restarted home assistant and added Adam through Plugwise beta:

log:
`2024-10-05 16:31:55.044 DEBUG (MainThread) [custom_components.plugwise] DUC cooldown interval: 1.5
2024-10-05 16:31:55.422 DEBUG (MainThread) [plugwise.constants] Plugwise identified as smile_open_therm_v3
2024-10-05 16:31:55.776 DEBUG (MainThread) [custom_components.plugwise] DUC update interval: 0:01:00
2024-10-05 16:31:56.125 DEBUG (MainThread) [custom_components.plugwise] Finished fetching plugwise data in 1.081 seconds (success: False) `
![Schermafbeelding 2024-10-05 om 16 36 06](https://github.com/user-attachments/assets/c8c19927-f583-452b-9824-2bfa99fc1f36)

**Comment by bouwew** [comment]
OK thanks.

Can you capture the full output from the `/core/domain_objects` webpage?
Browse to the URL, right-click anywhere in the page and the select Save as XML, if available? If not, please save as a txt-file.

If that succeeds, please send that file to my email: bouwe apestaartje westerdiijk punt info (er zelf even een geldig email adres van maken :) ).

**Comment by Torch1969** [comment]
> OK thanks.
> 
> Can you capture the full output from the `/core/domain_objects` webpage? If I have this file, I might discover pretty quickly what is causing the issue.
> 
> Browse to the URL, right-click anywhere in the page and the select Save as XML, if available? If not, please save as a txt-file.
> 
> If that succeeds, please send that file to my email: bouwe apestaartje westerdiijk punt info (er zelf even een geldig email adres van maken :) ).

Done, see mail

**Comment by bouwew** [comment]
The bug is caused by setting the type of the moved Plug (see top message) to "heater_central".
The Plugwise backend code assumed that this type was uniquely assigned to the Central heating boiler appliances.
Now we know that this is not correct.

**Comment by Torch1969** [comment]
For now I solved this “problem” by setting the type of the plug to something different (for instance energy meter). After that the integration and Adam hub are added without problem, also after switching back to core Plugwise integration. @bouwew thx for looking into this

**Comment by bouwew** [comment]
This bug is solved in the plugwise backend [v1.4.2](https://github.com/plugwise/python-plugwise/releases/tag/v1.4.2)

**Comment by bouwew** [comment]
@Torch1969 the fix for this issue will be available in Home Assistant 2024.11.0
